### DISCUSSION TITLE
mini.pick cursor

### DISCUSSION THREAD

**Discussion by skoch13** [nvim-mini/mini.nvim#614] [Q&A] [answered]
Hi, 
I'm a huge fan of your work and considering switching from Telescope. One thing that annoys me is the cursor shown in the focused line. Is there an option to turn it off?
![Screenshot2023-12-15 at 08 24 59](https://github.com/echasnovski/mini.nvim/assets/29177689/b67d72f0-db37-408d-8373-e6b443d548cb)

**Comment by echasnovski** [comment, marked as answer]
Hi :wave: 

It most certainly is a result of using 'folke/noice.nvim' alongside 'mini.pick'. See [this comment](https://github.com/echasnovski/mini.nvim/issues/513#issuecomment-1777115714).Without it, 'mini.pick' goes to some lengths for cursor to not appear in the picker window (by putting it in the Command line).

So, unfortunately, it seems that the only suggestion here is to stop using Noice.

**Reply by skoch13** [reply]
Thank you for your reply. Is it the same reason for not showing the dialog's titles or there is a setting to enable?

**Reply by echasnovski** [reply]
I am not sure I follow. What do you mean by "not showing dialog's titles"?

The reason this happens with 'noice.nvim' is described in the linked comment. Basically, the core culprit is the lack of good Neovim API to actually hide the cursor.

**Reply by skoch13** [reply]
Sorry for not being clear. In your demo each `Pick` dialog has a title down below describing the picker being used at the moment. Now, in my setup with noice there is no any title for dialogs opened via mini.pick

**Reply by echasnovski** [reply]
Ah, I see.

This utilizes [floating window footer](https://github.com/echasnovski/mini.nvim/blob/333d2d1090c80ac936b960469a6e93982cbaeb21/doc/mini-pick.txt#L53-L54) functionality which is only available on Neovim>=0.10. So it will automatically appear if you use new enough version. Right now 0.10 is the current nightly build which can be obtained through [Neovim's release](https://github.com/neovim/neovim/releases/tag/nightly) page. I recommend `appimage` as the most straightforward approach.

**Comment by lostl1ght** [comment]
I've found that changing and restoring the cursor shape with mini.pick's events works great.
Your terminal or GUI must support styling the cursor `:h 'guicursor'`.

![image](https://github.com/echasnovski/mini.nvim/assets/61948252/d779563a-ea43-490a-807e-f5b462829c19)


```lua
local guicursor = vim.opt.guicursor:get()
vim.api.nvim_create_autocmd('User', {
  pattern = 'MiniPickStart',
  callback = function() vim.opt.guicursor = { 'n:hor1' } end,
})
vim.api.nvim_create_autocmd('User', {
  pattern = 'MiniPickStop',
  callback = function() vim.opt.guicursor = guicursor end,
})
```

**Reply by echasnovski** [reply]
That's an interesting idea. Thanks for sharing!
